pkg/rancher-desktop/main/commandServer/__tests__/settingsValidator.spec.ts

```diff
@@ -316,64 +399,139 @@ describe(SettingsValidator, () => {
       });
     });
 
-    describe('kubernetes.version', () => {
-      it('should accept a valid version', () => {
-        const [needToUpdate, errors] = subject.validateSettings(cfg, { kubernetes: { version: '1.0.0' } });
+    it('should reject invalid values', () => {
+      const [needToUpdate, errors] = subject.validateSettings(cfg,
+        { application: { pathManagementStrategy: 'invalid value' as PathManagementStrategy } }, null);
 
-        expect({ needToUpdate, errors }).toEqual({
-          needToUpdate: true,
-          errors:       [],
-        });
+      expect({ needToUpdate, errors }).toEqual({
+        needToUpdate: false,
+        errors:       [`application.pathManagementStrategy: "invalid value" is not a valid strategy`],
       });
+    });
 
-      it('should reject an unknown version', () => {
-        const [needToUpdate, errors] = subject.validateSettings(cfg, { kubernetes: { version: '3.2.1', enabled: true } });
+    it('should reject setting as NotSet', () => {
+      const [needToUpdate, errors] = subject.validateSettings(cfg,
+        { application: { pathManagementStrategy: PathManagementStrategy.NotSet } }, null);
 
-        expect({ needToUpdate, errors }).toEqual({
-          needToUpdate: false,
-          errors:       [`Kubernetes version "3.2.1" not found.`],
-        });
+      expect({ needToUpdate, errors }).toEqual({
+        needToUpdate: false,
+        errors:       [`application.pathManagementStrategy: "notset" is not a valid strategy`],
       });
+    });
+  });
 
-      it('should normalize the version', () => {
-        const [needToUpdate, errors] = subject.validateSettings(
-          cfg,
-          { kubernetes: { version: 'v1.0.0+k3s12345' } });
+  describe('locked fields', () => {
+    describe('containerEngine.allowedImages', () => {
+      const allowedImageListConfig: settings.Settings = _.merge({}, cfg, {
+        containerEngine: {
+          allowedImages: {
+            enabled:  false,
+            patterns: ['pattern1', 'pattern2', 'pattern3'],
+          },
+        },
+      });
 
-        expect({ needToUpdate, errors }).toEqual({
-          needToUpdate: true,
-          errors:       [],
+      describe('when a field is locked', () => {
+        describe('locking allowedImages.enabled', () => {
+          const lockedSettings = { containerEngine: { allowedImages: { enabled: true } } };
+
+          it("locked allowedImages:enabled-field can't be changed", () => {
```

Comment:
Makes sense, although I never actually see this form of test output.

Reply:
Oh, interesting; I was just running `jest` (a.k.a. `npm run test:unit:jest`)… it's probably just because we only care about the failures most of the time :D